optaplanner-core/src/main/java/org/optaplanner/core/api/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScoreHolder.java

```diff
@@ -64,29 +64,32 @@ public void configureConstraintWeight(Rule rule, HardMediumSoftBigDecimalScore c
         super.configureConstraintWeight(rule, constraintWeight);
         BiConsumer<RuleContext, BigDecimal> matchExecutor;
         if (constraintWeight.equals(HardMediumSoftBigDecimalScore.ZERO)) {
-            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight) -> {};
-        } else if (constraintWeight.getMediumScore().equals(BigDecimal.ZERO) && constraintWeight.getSoftScore().equals(BigDecimal.ZERO)) {
-            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight)
-                    -> addHardConstraintMatch(kcontext, constraintWeight.getHardScore().multiply(matchWeight));
-        } else if (constraintWeight.getHardScore().equals(BigDecimal.ZERO) && constraintWeight.getSoftScore().equals(BigDecimal.ZERO)) {
-            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight)
-                    -> addMediumConstraintMatch(kcontext, constraintWeight.getMediumScore().multiply(matchWeight));
-        } else if (constraintWeight.getHardScore().equals(BigDecimal.ZERO) && constraintWeight.getMediumScore().equals(BigDecimal.ZERO)) {
-            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight)
-                    -> addSoftConstraintMatch(kcontext, constraintWeight.getSoftScore().multiply(matchWeight));
+            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight) -> {
+            };
+        } else if (constraintWeight.getMediumScore().equals(BigDecimal.ZERO)
+                && constraintWeight.getSoftScore().equals(BigDecimal.ZERO)) {
+            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight) -> addHardConstraintMatch(kcontext,
+                    constraintWeight.getHardScore().multiply(matchWeight));
+        } else if (constraintWeight.getHardScore().equals(BigDecimal.ZERO)
+                && constraintWeight.getSoftScore().equals(BigDecimal.ZERO)) {
+            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight) -> addMediumConstraintMatch(kcontext,
+                    constraintWeight.getMediumScore().multiply(matchWeight));
+        } else if (constraintWeight.getHardScore().equals(BigDecimal.ZERO)
+                && constraintWeight.getMediumScore().equals(BigDecimal.ZERO)) {
+            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight) -> addSoftConstraintMatch(kcontext,
+                    constraintWeight.getSoftScore().multiply(matchWeight));
         } else {
-            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight)
-                    -> addMultiConstraintMatch(kcontext,
+            matchExecutor = (RuleContext kcontext, BigDecimal matchWeight) -> addMultiConstraintMatch(kcontext,
```

Comment:
@rsynek Example of removing "manual" line break of different type. It's not allowed to break line around the arrow token.